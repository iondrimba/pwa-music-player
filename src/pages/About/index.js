import React, { PureComponent, Fragment } from 'react';
import { credits } from '../../data';
import './styles.scss';

class About extends PureComponent {
  constructor() {
    super();

    this.state = { credits };
  }

  render() {
    return (
      <Fragment>
        <div className="content">
          <h1 tabIndex="-1">PWA Music Player</h1>
          <p>This is a personal project built in my spare time for learning purposes.</p>
          <h2>Credits:</h2>
          <ul>
            {
              this.state.credits.map((license) => {
                return <li key={license.label}>{license.label} <a tabIndex="-1" target="_blank" rel="noopener noreferrer" href={license.link} >{license.author}</a></li>
              })
            }
          </ul>
        </div>

      </Fragment>
    );
  }
}

export default About;
