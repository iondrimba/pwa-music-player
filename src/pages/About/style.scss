.about {
  background-color: #fff;
  z-index: 3;

  .content {
    padding: 0 30px;
    height: 100%;
  }

  ul {
    list-style: none;
    padding: 0;

    li {
      margin: 10px 0;
    }
  }

  a {
    color: #7aa4e3;
    text-decoration: none;
    cursor: pointer;
  }
}
