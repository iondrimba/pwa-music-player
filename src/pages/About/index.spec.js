import React from 'react';
import renderer from 'react-test-renderer';
import About from '.';

describe('About', () => {
  it('renders without errors', () => {
    const component = renderer.create(<About/>);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
