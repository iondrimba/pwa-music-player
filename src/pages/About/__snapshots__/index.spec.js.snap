// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders without errors 1`] = `
<div
  className="content"
>
  <h1
    tabIndex="-1"
  >
    PWA Music Player
  </h1>
  <p>
    This is a personal project built in my spare time for learning purposes.
  </p>
  <h2>
    Credits:
  </h2>
  <ul>
    <li>
      Design inspired by
       
      <a
        href="https://dribbble.com/shots/3000629-Music-Player-Sketch-Freebie"
        rel="noopener noreferrer"
        tabIndex="-1"
        target="_blank"
      >
        Armas Nurbahari
      </a>
    </li>
    <li>
      Songs API by
       
      <a
        href="https://developers.soundcloud.com/docs/api"
        rel="noopener noreferrer"
        tabIndex="-1"
        target="_blank"
      >
        Soundcloud
      </a>
    </li>
    <li>
      Pause icon by
       
      <a
        href="https://www.flaticon.com/authors/smashicons"
        rel="noopener noreferrer"
        tabIndex="-1"
        target="_blank"
      >
        Smashicons
      </a>
    </li>
    <li>
      Left arrow icon by
       
      <a
        href="https://www.flaticon.com/authors/lucy-g"
        rel="noopener noreferrer"
        tabIndex="-1"
        target="_blank"
      >
        Lucy G
      </a>
    </li>
    <li>
      Play arrow icon by
       
      <a
        href="https://www.freepik.com/"
        rel="noopener noreferrer"
        tabIndex="-1"
        target="_blank"
      >
        Freepik
      </a>
    </li>
    <li>
      Help icon by
       
      <a
        href="https://www.flaticon.com/authors/anton-saputro"
        rel="noopener noreferrer"
        tabIndex="-1"
        target="_blank"
      >
        Anton Saputro
      </a>
    </li>
    <li>
      Close icon by
       
      <a
        href="https://www.flaticon.com/authors/chanut"
        rel="noopener noreferrer"
        tabIndex="-1"
        target="_blank"
      >
        Chanut
      </a>
    </li>
  </ul>
</div>
`;
